core/src/main/java/org/elasticsearch/search/sort/ScriptSortBuilder.java

```diff
@@ -216,6 +217,32 @@ public XContentBuilder toXContent(XContentBuilder builder, Params builderParams)
         return builder;
     }
 
+    private static ConstructingObjectParser<ScriptSortBuilder, QueryParseContext> PARSER = new ConstructingObjectParser<>(NAME,
+            a -> new ScriptSortBuilder((Script) a[0], (ScriptSortType) a[1]));
+
+    static {
+        PARSER.declareField(constructorArg(), (p, c) -> {
+            try {
+                return Script.parse(p, c.getParseFieldMatcher(), c.getDefaultScriptLanguage());
+            } catch (IOException e) {
+                throw new ParsingException(p.getTokenLocation(), "Error parsing [" + ScriptField.SCRIPT.getPreferredName() + "] field", e);
+            }
+        }, ScriptField.SCRIPT, ValueType.OBJECT_OR_STRING);
+        PARSER.declareField(constructorArg(), p -> ScriptSortType.fromString(p.text()), TYPE_FIELD, ValueType.STRING);
+        PARSER.declareField(ScriptSortBuilder::order, p -> SortOrder.fromString(p.text()), ORDER_FIELD, ValueType.STRING);
+        PARSER.declareField(ScriptSortBuilder::sortMode, p -> SortMode.fromString(p.text()), SORTMODE_FIELD, ValueType.STRING);
```

Comment:
I wonder if `PARSER.declareString((b, v) -> b.sortMode(SortMode.fromString(b), SORTMODE_FIELD);` is better? I kind of prefer it because then you don't need to think about `ValueType`.